anchore_engine/utils.py

```diff
@@ -858,3 +859,53 @@ def bytes_to_mb(value, round_to=None):
         mb = round(mb, round_to)
 
     return mb
+
+
+def dedup_list_of_dicts(list_of_dicts: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """
+    This helper function will sort and dedup a list of dicts. Dicts are not hashable, so you can't just call set on
+    them. This will recursively convert nested dictionaries to frozensets, add them to a set, recursively
+    convert the frozensets back to dictionaries, and convert the set to a list.
+
+    :param list_of_dicts: list of dictionaries to deduplicate
+    :type list_of_dicts: List[Dict[str, Any]]
+    :return: deduplicated list of dictionaries
+    :rtype: List[Dict[str, Any]]
+    """
+    unique_items = set(_recursive_freeze(child_dict) for child_dict in list_of_dicts)
+    return [
+        _recursive_unfreeze(child_frozenset) for child_frozenset in sorted(unique_items)
+    ]
```

Comment:
This is a generic approach that works with any data shape. Are there any other solutions that might make it simpler and or more performant? One solution that comes to my mind is using named tuples as they are hashable and have faster creation (not read) times than dataclasses or objects. They might also be better suited for this data since it's not nested.  Granted, it's tied to the data shape and not as generic as the approach here. 

Pseudo-code for named tupel approach: create an output map/dict where the key is the named tuple and value is the affected package dictionary. Iterate over the input list and create a named tuple for each item, check if it's present in the map before adding it. Return the list-ified values of the output. 
```
AffectedPackageTuple = namedTuple("AffectedPackage", ["name", "type", "version", "wont_fix"])
output_map = dict()
for item in list_of_dicts:
  # create the named tuple using item
  # check if the named tuple is present in the output map and
  # add the named tuple-dict key-value pair to the output map
return list(output_map.values())
```

Reply:
tried some different things and went with simplified frozenset cast. this approach has the best average time for dedup. see latest commit